src/Term_Command.php

```diff
@@ -260,11 +268,22 @@ public function create( $args, $assoc_args ) {
 	 *     # Get details about a category with id 199.
 	 *     $ wp term get category 199 --format=json
 	 *     {"term_id":199,"name":"Apple","slug":"apple","term_group":0,"term_taxonomy_id":199,"taxonomy":"category","description":"A type of fruit","parent":0,"count":0,"filter":"raw"}
+	 *
+	 *     # Get details about a category with slug apple.
+	 *     $ wp term get category apple --by=slug --format=json
+	 *     {"term_id":199,"name":"Apple","slug":"apple","term_group":0,"term_taxonomy_id":199,"taxonomy":"category","description":"A type of fruit","parent":0,"count":0,"filter":"raw"}
 	 */
 	public function get( $args, $assoc_args ) {
 
-		list( $taxonomy, $term_id ) = $args;
-		$term = get_term_by( 'id', $term_id, $taxonomy );
+		list( $taxonomy, $term ) = $args;
+
+		// Get term by specified argument otherwise get term by id.
+		if ( $field = Utils\get_flag_value( $assoc_args, 'by' ) ) {
+			$term = get_term_by( $field, $term, $taxonomy );
+		} else {
+			$term = get_term_by( 'id', $term, $taxonomy );
+		}
```

Comment:
With `default: id` defined above, instead of the conditional you can have:

```
$term = get_term_by( Utils\get_flag_value( $assoc_args, 'by' ), $term, $taxonomy );
```